src/tools/ceph_objectstore_tool.cc

```diff
@@ -2501,6 +2501,43 @@ int print_obj_info(ObjectStore *store, coll_t coll, ghobject_t &ghobj, Formatter
   return r;
 }
 
+int corrupt_info(ObjectStore *store, coll_t coll, ghobject_t &ghobj, Formatter* formatter)
+{
+  auto ch = store->open_collection(coll);
+  bufferlist attr;
+  int r = store->getattr(ch, ghobj, OI_ATTR, attr);
+  if (r < 0) {
+    cerr << "Error getting attr on : " << make_pair(coll, ghobj) << ", "
+       << cpp_strerror(r) << std::endl;
+    return r;
+  }
+  object_info_t oi;
+  auto bp = attr.cbegin();
+  try {
+    decode(oi, bp);
+  } catch (...) {
+    r = -EINVAL;
+    cerr << "Error getting attr on : " << make_pair(coll, ghobj) << ", "
+         << cpp_strerror(r) << std::endl;
+    return r;
+  }
+  if (!dry_run) {
+    attr.clear();
+    oi.alloc_hint_flags += 0xff;
```

Comment:
shoudn't this corrupt the data_digest instead?

Reply:
The purpose of his change is to have 2 errors, object_info_inconsistency AND data_digest_mismatch_info.  That way we see that both kinds of repairs happen during a repair.  One uses recovery to fix and the other missing_digest mechanism.

Reply:
oh i see, and the test is updating the bytes separately via a different cot call. great!